db/write_callback_test.cc

```diff
@@ -184,7 +190,8 @@ TEST_F(WriteCallbackTest, WriteWithCallbackTest) {
                       ASSERT_TRUE(writer->callback->Callback(nullptr).ok() ==
                                   !write_group.back().callback_.should_fail_);
                     }
-
+                    
+                    threads_waiting.fetch_add(1);
```

Comment:
Previously the callback is like:
S1. bump threads_waiting
S2. a couple of asserts
S3. wait for threads_waiting = write_group.size().
Leader can exit the callback after followers finish S1 but not S2. By swapping S1 and S2, follower must finish S2 before leader can finish S3.